src/sema/builtin.rs

```diff
@@ -1122,6 +1135,85 @@ pub fn resolve_namespace_call(
                 return Err(());
             }
         }
+        Builtin::AbiEncodeCall => {
+            // first argument is function
+            if let Some(function) = args_iter.next() {
+                let function = expression(
+                    function,
+                    context,
+                    ns,
+                    symtable,
+                    diagnostics,
+                    ResolveTo::Unknown,
+                )?;
+
+                match function.ty() {
+                    Type::ExternalFunction { params, .. }
+                    | Type::InternalFunction { params, .. } => {
+                        resolved_args.push(function);
+
+                        if args.len() - 1 != params.len() {
+                            diagnostics.push(Diagnostic::error(
+                                *loc,
+                                format!(
+                                    "function takes {} arguments, {} provided",
+                                    params.len(),
+                                    args.len() - 1
```

Comment:
`abi.encodeCall` receives a function as the first argument. Does `params.len()` takes that into account? I think it would be confusing if you don't count the function as an argument in the error message.

Reply:
`params` is a reference to the function prototype parameters, and `args` is the arguments provided to `abi.encodeCall()`. I think this is right.

Reply:
When the function is properly invoked `args.len() == params.len()` should be true, right? You compare `args.len()-1` with `params.len()`.

Reply:
Take this example:
```
    contract c {
        function f1() public {
            bytes foo = abi.encodeCall(c.bar, 102, true);
        }
        function bar(int a, bool b) public {}
    }
```

The call to `abi.encodeCall()` has three arguments: the function, and the two arguments for the function.